.travis.yml

```
@@ -1,44 +1,51 @@
-dist: xenial
-language: python
-python:
-  - '3.8'
-os: 
-  - linux
+dist: bionic
+language: generic
 
 addons:
   apt:
     packages:
       - git
       - wget
+      - flex
+      - bison
       - make
       - gperf
-      - python
+      - python3
+      - python3-pip
+      - python3-setuptools
       - python-serial
-      - python-future
+      - python-click
       - python-cryptography
+      - python-future
       - python-pyparsing
+      - python-pyelftools
+      - cmake
+      - ninja-build
+      - ccache
+      - libffi-dev
+      - libssl-dev
       - libncurses-dev
-      - flex
-      - bison
+      - doxygen
 
 
 before_install:
-- sudo apt-get -y install doxygen
+# - sudo apt-get -y install doxygen
 # Save path to the git repository
 - PROJECT_PATH=$(pwd)
 
 install:
-- pip install -U codespell
-- pip install -U travis-sphinx
-- pip install -U sphinx-rtd-theme
-- pip install -U breathe
-- pip install -U exhale
+- pip3 install -U codespell
+- pip3 install -U travis-sphinx
+- pip3 install -U sphinx-rtd-theme
+- pip3 install -U breathe
+- pip3 install -U exhale
 # Install ESP32 toochain following steps as described
 # in https://docs.espressif.com/projects/esp-idf/en/latest/get-started/#get-started-get-prerequisites
 - mkdir -p ~/esp
 - cd ~/esp
 - git clone --recursive https://github.com/espressif/esp-idf.git
 - cd ~/esp/esp-idf
+# - git checkout release/v4.0
```

```suggestion
- git clone --recursive https://github.com/espressif/esp-idf.git -b release/v4.1
- cd ~/esp/esp-idf
```

That way release 4.1 (latest stable one) can be used. Saves one line too :tada:

---

Oops, my bad, it's v4.0

---

the latest release is v4.0